### PR TITLE
System.io.file system.access control

### PR DESCRIPTION
**Pull request by dsgouda**
@weshaggard

### CODE REVIEW

File: src/System.IO.FileSystem.AccessControl/src/Interop/Interop.Manual.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+internal partial class Interop
```
Comment (weshaggard) [inline src/System.IO.FileSystem.AccessControl/src/Interop/Interop.Manual.cs]: These should be combined into the common interop directory. I suspect a lot of these already exist there. 
See https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Windows/mincore/Interop.Errors.cs as an example.

File: src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.builds [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup>
+    <Project Include="System.IO.FileSystem.AccessControl.csproj">
+      <AdditionalProperties>TargetGroup=</AdditionalProperties>
+    </Project>
+    <Project Include="System.IO.FileSystem.AccessControl.csproj">
```
Comment (weshaggard) [inline src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.builds]: For netcore50 and net46 add the TargetsWindows=true condition.

File: src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  <ItemGroup Condition="'$(TargetGroup)'!='net46'">
+    <Compile Include="Interop\Interop.Manual.cs" />
+    <Compile Include="System\Security\AccessControl\FileSecurity.cs" />
+    <Compile Include="System\Security\AccessControl\DirectoryObjectSecurity.cs" />
+    <Compile Include="System\IO\FileSystemAclExtensions.cs" />
+    <Compile Include="$(CommonPath)\System\NotImplemented.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)'=='net46'">
+    <TargetingPackReference Include="mscorlib" />
+    <Compile Include="System\IO\FileSystemAclExtensions.Desktop.cs" />
```
Comment (weshaggard) [inline src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: Please rename this file to .net46 instead of .Desktop.

File: src/System.Runtime/System.Runtime.sln [hunk 4]
```diff
@@ -9,7 +9,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".nuget", ".nuget", "{1CA59C
 		..\.nuget\packages.Windows_NT.config = ..\.nuget\packages.Windows_NT.config
 	EndProjectSection
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Runtime.CoreCLR", "src\System.Runtime.CoreCLR.csproj", "{56B9D0A9-44D3-488E-8B42-C14A6E30CAB2}"
```
Comment (weshaggard) [inline src/System.Runtime/System.Runtime.sln]: These System.Runtime changes look unrelated to this change.

Comment (dsgouda) [inline src/System.Runtime/System.Runtime.sln]: I was hoping not to create a separate PR for a minor change to .sln file here.

**Comment by dsgouda** [comment]
Addressed all comments. Opening new PR at dotnet master
